ver_linux.go

```diff
@@ -20,98 +18,89 @@ import (
 //		- r.nix.Kernel
 //		- r.nix.Distro
 //		- r.nix.PkgMng
-func GetVersion() *Release {
-
-	inf := &Release{
+func GetVersion() Release {
+	info := Release{
 		Runtime: runtime.GOOS,
 		Arch:    runtime.GOARCH,
+		Name:    "unknown",
+		Version: "unknown",
+
+		Linux: LinuxRelease{
+			Distro:     "unknown",
+			Kernel:     "unknown",
+			PkgManager: "unknown",
+		},
 	}
 
-	out, _ := exec.Command("uname", "-r").Output()
-	kern := strings.ReplaceAll(strings.ReplaceAll(string(out), "\n", ""), "\"", "")
-	inf.nix.Kernel = string(kern)
-
-	if pathExists("/etc/os-release") == true {
-		f := readFile("/etc/os-release")
-
-		var (
-			nameField  = regexp.MustCompile(`NAME=(.*?)\n|\nNAME=(.*?)\n`)
-			pnameField = regexp.MustCompile(`PRETTY_NAME=(.*?)\n|\nPRETTY_NAME=(.*?)\n`)
-			verField   = regexp.MustCompile(`VERSION_ID=(.*?)\n|\nVERSION_ID=(.*?)\n`)
-
-			suse   = regexp.MustCompile(`SLES|openSUSE`)
-			debian = regexp.MustCompile(`Debian|Ubuntu|Kali|Parrot|Mint`)
-			rhl    = regexp.MustCompile(`Red Hat|CentOS|Fedora|Oracle`)
-			arch   = regexp.MustCompile(`Arch|Manjaro`)
-			alpine = regexp.MustCompile(`Alpine`)
-		)
-
-		namef := strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(nameField.FindString(f), "\n", ""), "\"", ""))
-		pnamef := strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(pnameField.FindString(f), "\n", ""), "\"", ""))
-		verf := strings.TrimSpace(strings.ReplaceAll(strings.ReplaceAll(verField.FindString(f), "\n", ""), "\"", ""))
-
-		name := strings.Split(namef, "=")[1]
-
-		if pnamef != "" {
-			inf.Name = strings.Split(pnamef, "=")[1]
-		} else {
-			inf.Name = "unknown"
-		}
-
-		if verf != "" {
-			inf.Version = strings.Split(verf, "=")[1]
-		} else {
-			inf.Version = "unknown"
-		}
-
-		if suse.MatchString(name) == true {
-			inf.nix.Distro = "opensuse"
-			inf.nix.PkgMng = "zypper"
-
-		} else if debian.MatchString(name) == true {
-			inf.nix.Distro = "debian"
-			inf.nix.PkgMng = "apt"
-
-		} else if rhl.MatchString(name) == true {
-			inf.nix.Distro = "fedora"
-			inf.nix.PkgMng = "yum"
-
-		} else if arch.MatchString(name) == true {
-			inf.Version = "rolling"
-			inf.nix.Distro = "arch"
-			inf.nix.PkgMng = "pacman"
-
-		} else if alpine.MatchString(name) == true {
-			inf.nix.Distro = "alpine"
-			inf.nix.PkgMng = "apk"
-
-		} else {
-			inf.nix.Distro = "unknown"
-			inf.nix.PkgMng = "unknown"
-		}
+	out, err := exec.Command("uname", "-r").Output()
+	if err != nil {
+		return info
```

Comment:
If it can't get the kernel version, it should still continue on and try to read `/etc/os-release`

Reply:
ok